app/controllers/carts_controller.rb

```diff
@@ -1,20 +1,35 @@
 class CartsController < ApplicationController
 
+  before_action :find_cart
+
   def add
     dish = Dish.find(params[:id])
-    quantity = params[:amount].nil? ? 1 : params[:amount]
-    if session[:cart_id]
-      show
+    if an_integer?(params[:amount])
```

Comment:
I'm not sure I understand. Isn't it enough to use 'if params[:amount]'? If you want to hedge yourself from users passing in letters you need to add an error handler and feedback in UI. Not sure if that is part of this feature though.

Reply:
Disregard from this comment. I read the acceptance test and you are doing everything 100% right. I was a bit unfocused...